### DISCUSSION TITLE
Why would this.parent() in a subdocument not have fields sometimes?

### DISCUSSION THREAD

**Discussion by eliw00d** [Automattic/mongoose#10092] [Q&A]
With Mongoose 5.10.11...

We are using a default function in a subdocument to get a field from `this.parent()` but sometimes the value is undefined - despite being in MongoDB itself. Why would that happen? We are not using lean or toObject.

Example:
```javascript
child: {
  type: [
    {
      field: {
        type: String,
        default() {
          return this.parent().field // undefined for some documents despite having a defined value in MongoDB
        },
      },
    },
  ],
},
field: {
  type: String,
},
```

What's really strange is that if I add a default to the the parent field we are trying to use, that does get used!

Example:
```javascript
child: {
  type: [
    {
      field: {
        type: String,
        default() {
          return this.parent().field // "Default for Field" for some documents despite having a defined value in MongoDB
        }
      }
    }
  ]
},
field: {
  type: String,
  default: 'Default for Field'
},
```

**Comment by vkarpov15** [comment]
In theory there shouldn't be any reason for this. If you can open up an issue and post a code snippet that reproduces this issue, we'd be happy to fix it :+1:
